src/batou_ext/nix.py

```diff
@@ -102,37 +102,36 @@ class UserEnv(batou.component.Component):
                     "yarn",
                     "zip",
                 ],
-                channel="<nixos-channel-base-url-to-be-put-in>/nixexprs.tar.xz"
                 shellInit="# additional shell init")
 
     A list of available channels can be found at e.g.
     https://nixos.org/channels/
     """
 
     namevar = "profile_name"
-    channel = batou.component.Attribute(str)
     shellInit = ""
     packages = ()
     let_extra = ""
+    channel = None
 
     def configure(self):
-        self.checksum = hashlib.sha256()
+        if self.channel:
+            batou.output.step(
+                "WARNING",
+                "using `channel` with UserEnv is deprecated, please remove it from your deployment code.",
+                yellow=True,
+            )
+
         template = pkg_resources.resource_string(
             __name__, "resources/userenv.nix"
         ).decode("UTF-8")
         self.profile = self.expand(template).encode("UTF-8")
-        self.checksum.update(self.profile)
-        self.nix_env_name = self.expand(
-            "{{component.profile_name}}-1.{{component.checksum.hexdigest()}}"
-        )
+
         self += batou.lib.file.File(
-            self.profile_name + ".nix", content=self.profile
-        )
-        self += Package(self.nix_env_name, file=self._.path)
```

Comment:
What's the migration scenario? I don't see the nix-env-env being removed/uninstalled. So it'll lie around indefinitely and probably still has precedence over the new env.

Reply:
Currently it would rely on manual intervention. Uninstalling all nix-envs might quickly lead to unexpected behavior and there is no good way to determine which envs were installed by this component previously. The env's name could change at any time.